iothub_client/src/iothubtransport_amqp_twin_messenger.c

```diff
@@ -2115,6 +2115,11 @@ int twin_messenger_stop(TWIN_MESSENGER_HANDLE twin_msgr_handle)
         }
         else
         {
+            if (twin_msgr->subscription_state != TWIN_SUBSCRIPTION_STATE_UNSUBSCRIBE)
```

Comment:
There is no comment added in this PR clarifying the purpose of this check, and it's not initially clear without knowing the GH Issue in reference. Can you add a comment specifying why this if statement and set is necessary?

Reply:
Sure, please read the description, I added the details there.

Reply:
yes but the description will not be in the code :) this is for future developer sanity. One line will do:

// if previously subscribed, twin_messenger_stop moves the subscription state to resubscribe automatically to receive twin updates

Reply:
I'll add the requirement docs.